### PR TITLE
Include new reactor migration as of springboot 3.2

### PR DESCRIPTION
**Pull request by Laurens-W**
Spring boot upgrade to 3.2.4 also upgrades io.projectreactor/reactor-bom from 2020.0.27 to 2023.0.4. This in turn upgrades reactor-core module from 3.4.26 to 3.6.5.

## Anyone you would like to review specifically?

@timtebeek 

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

**Comment by timtebeek** [comment]
Something odd is going on with the required versions here ; likely Spring Boot silently had a breaking version upgrade:
SB 3.1.0 came with Reactor BOM 2020.0.7, which manages reactor-core 3.4.6
https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-dependencies/3.1.0/spring-boot-dependencies-3.1.0.pom
https://repo1.maven.org/maven2/io/projectreactor/reactor-bom/2020.0.7/reactor-bom-2020.0.7.pom

SB 3.1.12 came with Reactor BOM 2020.0.19, which manages reactor-core 3.5.17
https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-dependencies/3.1.12/spring-boot-dependencies-3.1.12.pom
https://repo1.maven.org/maven2/io/projectreactor/reactor-bom/2022.0.19/reactor-bom-2022.0.19.pom

So in summary: the breaking change is already in 3.1.x, since we migrate folks to the latest in that range. I'll move accordingly.

**Review by timtebeek — APPROVED** [review]
Thanks a lot! Good not to miss these necessary steps going to 3.1+.
